examples/petsc/Makefile

```diff
@@ -1,4 +1,5 @@
 PETSc.pc := $(PETSC_DIR)/$(PETSC_ARCH)/lib/pkgconfig/PETSc.pc
+PETSc.pc := $(or $(wildcard $(PETSc.pc)),$(PETSC_DIR)/lib/pkgconfig/PETSc.pc)
```

Comment:
In what circumstance is there a `PETSc.pc` in this directory? If we really need to add this, we can delete the line above.

Reply:
The location `$(PETSC_DIR)/lib/pkgconfig/PETSc.pc` is what I need when using PETSc installed through spack, for example.

Reply:
Yes, but the change is to look for `PETSc.pc` in the current directory. Why that change?

Reply:
Whoops, my mistake. So the environment you're thinking of is that `PETSC_ARCH` is set for a prefix install where it should not be?

Reply:
The intent is to support installations in either `$(PETSC_DIR)/$(PETSC_ARCH)` or in `$(PETSC_DIR)`.

I guess the line I added to support the latter is unnecessary if `PETSC_ARCH` is undefined or empty - the path will have a double slash but that should still work. I probably looked-up where `PETSc.pc` is (in a spack install) without looking at the first case and never realized that leaving `PETSC_ARCH` empty is an option.

I'll remove the added line - I'll just add a comment that `PETSC_ARCH` can be empty.

Reply:
Perfect, thanks!